RecoTracker/FinalTrackSelectors/plugins/SingleLongTrackProducer.cc

```diff
@@ -0,0 +1,221 @@
+// user includes
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+// ROOT includes
+#include "TLorentzVector.h"
+
+class SingleLongTrackProducer : public edm::stream::EDProducer<> {
+public:
+  explicit SingleLongTrackProducer(const edm::ParameterSet &);
+  ~SingleLongTrackProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  void produce(edm::Event &, const edm::EventSetup &) override;
+
+  const edm::EDGetTokenT<std::vector<reco::Track>> tracksToken;
+  const edm::EDGetTokenT<std::vector<reco::Muon>> muonsToken;
+  const edm::EDGetTokenT<reco::VertexCollection> PrimVtxToken;
+
+  const int minNumberOfLayers;
+  const double matchInDr;
+  const bool onlyValidHits;
+  const bool debug;
+  const double minPt;
+  const double maxEta;
+  const double maxDxy;
+  const double maxDz;
+};
+
+SingleLongTrackProducer::SingleLongTrackProducer(const edm::ParameterSet &iConfig)
+    : tracksToken{consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("allTracks"))},
+      muonsToken{consumes<std::vector<reco::Muon>>(iConfig.getParameter<edm::InputTag>("matchMuons"))},
+      PrimVtxToken{consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("PrimaryVertex"))},
+      minNumberOfLayers{iConfig.getParameter<int>("minNumberOfLayers")},
+      matchInDr{iConfig.getParameter<double>("requiredDr")},
+      onlyValidHits{iConfig.getParameter<bool>("onlyValidHits")},
+      debug{iConfig.getParameter<bool>("debug")},
+      minPt{iConfig.getParameter<double>("minPt")},
+      maxEta{iConfig.getParameter<double>("maxEta")},
+      maxDxy{iConfig.getParameter<double>("maxDxy")},
+      maxDz{iConfig.getParameter<double>("maxDz")} {
+  produces<reco::TrackCollection>("").setBranchAlias("");
```

Comment:
Would it make sense to set some name to this new collection? Something like minNumberOfLayers10 or any meaningful name in view of future collections? Or it is not expected to add more collections of this type? Thanks